skopt/optimizer/optimizer.py

```diff
@@ -175,28 +177,44 @@ def __init__(self, dimensions, base_estimator,
     def _check_arguments(self, base_estimator, n_initial_points,
                          acq_optimizer):
         """Check arguments for sanity."""
+
+        if isinstance(base_estimator, str):
+            base_estimator = cook_skopt_estimator(
+                base_estimator, random_state=self.rng,
+                n_dims=self.space.transformed_n_dims,
+                is_cat=self.space.is_categorical)
         if not is_regressor(base_estimator):
-            raise ValueError(
-                "%s has to be a regressor." % base_estimator)
-        self.base_estimator = base_estimator
+            raise ValueError("%s has to be a regressor." % base_estimator)
+        self.base_estimator_ = base_estimator
 
         if n_initial_points < 0:
             raise ValueError(
                 "Expected `n_initial_points` >= 0, got %d" % n_initial_points)
         self._n_initial_points = n_initial_points
         self.n_initial_points_ = n_initial_points
 
-        if (isinstance(base_estimator, (ExtraTreesRegressor,
-                                        RandomForestRegressor,
-                                        GradientBoostingQuantileRegressor)) and
-                not acq_optimizer == "sampling"):
-            raise ValueError(
-                "The tree-based regressor {0} should run with "
-                "acq_optimizer='sampling'".format(type(base_estimator)))
+        tree_ests = (
```

Comment:
-> `tree_estimators` or should it be something else because for a GP with categoricals we also use sampling right? `piecewise_constant_estimators` ... too long :-/